code/modules/mob/living/basic/guardian/guardian_types/explosive.dm

```diff
@@ -0,0 +1,77 @@
+/// A durable guardian which can convert objects into hidden explosives.
+/mob/living/basic/guardian/explosive
+	guardian_type = GUARDIAN_EXPLOSIVE
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	damage_coeff = list(BRUTE = 0.6, BURN = 0.6, TOX = 0.6, CLONE = 0.6, STAMINA = 0, OXY = 0.6)
+	range = 13
+	playstyle_string = span_holoparasite("As an <b>explosive</b> type, you have moderate close combat abilities and are capable of converting nearby items and objects into disguised bombs via right-click.")
+	creator_name = "Explosive"
+	creator_desc = "High damage resist and medium power attack. Can turn any object, including objects too large to pick up, into a bomb, dealing explosive damage to the next person to touch it. The object will return to normal after the trap is triggered or after a delay."
+	creator_icon = "explosive"
+	/// Ability which plants bombs
+	var/datum/action/cooldown/mob_cooldown/explosive_booby_trap/bomb
+
+/mob/living/basic/guardian/explosive/Initialize(mapload, datum/guardian_fluff/theme)
+	. = ..()
+	bomb = new(src)
+	bomb.Grant(src)
+
+/mob/living/basic/guardian/explosive/Destroy()
+	QDEL_NULL(bomb)
+	return ..()
+
+/mob/living/basic/guardian/explosive/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	if(LAZYACCESS(modifiers, RIGHT_CLICK) && proximity_flag && isobj(attack_target))
+		bomb.Trigger(target = attack_target)
+		return
+	return ..()
+
+
+/// An ability which can turn an object into a bomb
+/datum/action/cooldown/mob_cooldown/explosive_booby_trap
+	name = "Explosive Trap"
+	desc = "Convert an inanimate object into a deadly and mostly undetectable explosive, triggered on touch."
+	button_icon = 'icons/mob/actions/actions_spells.dmi'
+	button_icon_state = "smoke"
+	cooldown_time = 20 SECONDS
+	melee_cooldown_time = 0 SECONDS
+	background_icon = 'icons/hud/guardian.dmi'
+	background_icon_state = "base"
+	/// After this amount of time passses, bomb deactivates.
+	var/decay_time = 1 MINUTES
+	/// Static list of signals that activate the bomb.
+	var/static/list/boom_signals = list(COMSIG_ATOM_ATTACKBY, COMSIG_ATOM_BUMPED, COMSIG_ATOM_ATTACK_HAND)
+
+/datum/action/cooldown/mob_cooldown/explosive_booby_trap/PreActivate(atom/target)
+	if (!isobj(target))
+		return FALSE
+	if (!owner.Adjacent(target))
+		return FALSE
+	return ..()
+
+/datum/action/cooldown/mob_cooldown/explosive_booby_trap/Activate(atom/target)
+	target.balloon_alert(owner, "bomb planted")
```

Comment:
i feel like this balloon alert should be near the end of the proc before the cooldown just in case something really wonky happens